src/controllers/secret.go

```diff
@@ -0,0 +1,194 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dana-team/platform-backend/src/types"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretController interface {
+	CreateSecret(types.CreateSecretRequest) (types.CreateSecretResponse, error)
+	GetSecrets(types.GetSecretsRequest) (types.GetSecretsResponse, error)
+	GetSecret(types.GetSecretRequest) (types.GetSecretResponse, error)
+	PatcnSecret(types.UpdateSecretRequest) (types.UpdateSecretResponse, error)
+	DeleteSecret(types.DeleteSecretRequest) (types.DeleteSecretResponse, error)
+}
+
+type secretController struct {
+	client *kubernetes.Clientset
+	ctx    context.Context
+	logger *zap.Logger
+}
+
+func NewSecretController(client *kubernetes.Clientset, context context.Context, logger *zap.Logger) SecretController {
```

Comment:
Add [doc comments](https://google.github.io/styleguide/go/decisions#doc-comments) above each function

Reply:
I really believe in [self-documenting code](https://en.wikipedia.org/wiki/Self-documenting_code), unless we're planning to generate a new documentation from these comments.

Comments are really useful when we're exposing some new Kubernetes API [1] or writing some library that is used by others. In that case, it should be documented automatically.

Back to your request, when I see `NewSecretController` function, it's easy to understand from the name of the function and what's inside that it's returns a new `secretController`.

[1] https://github.com/dana-team/container-app-operator/blob/main/api/v1alpha1/capp_types.go#L30-L62

Reply:
I absolutely agree that clear and good code usually does not require many comments, but a short sentence above each function is still a good practice I'd like our codebase to adhere to.